File: branches/BibleWebService_trunk_2.0.0.0_CSharpConversion/BibleTranslations.vb
Imports System.Data.SqlClient
Imports System.Configuration.ConfigurationSettings
Imports System.Configuration

Public Class BibleTranslations
    Inherits CollectionBase

    Default Public Property Item(ByVal index As Integer) As BibleTranslation
        Get
            Return CType(list(index), BibleTranslation)
        End Get
        Set(ByVal Value As BibleTranslation)
            list(index) = Value
        End Set
    End Property

    Public Function Add(ByVal bt As BibleTranslation) As Integer
        Return list.Add(bt)
    End Function

    Public Sub Remove(ByVal bt As BibleTranslation)
        list.Remove(bt)
    End Sub

    Public Function Find() As Boolean
		Dim dtr As SqlDataReader = Nothing

        Try
            Dim strSQL As String = "bible_GetTranslations"
			Dim cnn As New SqlConnection(ConfigurationManager.AppSettings("DataConn"))
            Dim cmd As New SqlCommand(strSQL, cnn)
            Dim blnHasRows As Boolean

            cmd.CommandType = CommandType.StoredProcedure

            cnn.Open()
            dtr = cmd.ExecuteReader(CommandBehavior.CloseConnection)

            Do While dtr.Read
                Dim bt As New BibleTranslation

                bt.BibleNo = dtr.GetInt32(0)
                bt.BibleAbbr = dtr.GetString(1)
                bt.BibleName = dtr.GetString(2)

                list.Add(bt)
            Loop

            blnHasRows = dtr.HasRows
            dtr.Close()
            Return blnHasRows
		Catch sx As SqlException
			If dtr IsNot Nothing Then
				If Not dtr.IsClosed Then
					dtr.Close()
				End If
			End If

			Dim bt As New BibleTranslation
			bt.BibleNo = -1
			bt.BibleName = sx.ToString
			List.Add(bt)
			Return False
        Catch ex As Exception
            If Not dtr.IsClosed Then
                dtr.Close()
            End If

            Dim bt As New BibleTranslation
            bt.BibleNo = -1
            bt.BibleName = ex.ToString
            list.Add(bt)
            Return False
        End Try
    End Function
End Class

File: branches/BibleWebService_trunk_2.0.0.0_CSharpConversion/BibleTranslations.cs
using System;
using System.Collections;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;

namespace BibleWebService
{
    public class BibleTranslations : CollectionBase
    {
        public BibleTranslation this[int index]
        {
            get { return this[index]; }
            set { this[index] = value; }
        }

        //public BibleTranslation Item {
        //    get { return (BibleTranslation)list(index); }
        //    set { list(index) = value; }
        //}

        public int Add(BibleTranslation bt)
        {
            return Add(bt);
        }

        public void Remove(BibleTranslation bt)
        {
            Remove(bt);
        }

        public bool Find()
        {
            SqlDataReader dtr = null;

            try
            {
                string strSQL = "bible_GetTranslations";
                SqlConnection cnn = new SqlConnection(ConfigurationManager.AppSettings["DataConn"]);
                SqlCommand cmd = new SqlCommand(strSQL, cnn);
                bool blnHasRows;

                cmd.CommandType = CommandType.StoredProcedure;

                cnn.Open();
                dtr = cmd.ExecuteReader(CommandBehavior.CloseConnection);

                while (dtr.Read())
                {
                    BibleTranslation bt = new BibleTranslation();

                    bt.BibleNo = dtr.GetInt32(0);
                    bt.BibleAbbr = dtr.GetString(1);
                    bt.BibleName = dtr.GetString(2);

                    Add(bt);
                }

                blnHasRows = dtr.HasRows;
                dtr.Close();
                return blnHasRows;
            }
            catch (SqlException sx)
            {
                if (dtr != null)
                {
                    if (!dtr.IsClosed)
                    {
                        dtr.Close();
                    }
                }

                BibleTranslation bt = new BibleTranslation();
                bt.BibleNo = -1;
                bt.BibleName = sx.ToString();
                List.Add(bt);
                return false;
            }
            catch (Exception ex)
            {
                if (dtr != null)
                    if (!dtr.IsClosed)
                    {
                        dtr.Close();
                    }

                BibleTranslation bt = new BibleTranslation();
                bt.BibleNo = -1;
                bt.BibleName = ex.ToString();
                Add(bt);
                return false;
            }
        }
    }
}